tests/integration/_conftest/protostar_project_fixture.py

```diff
@@ -98,6 +99,22 @@ def create_contracts(self, contract_name_to_file: Dict[str, Union[str, Path]]):
             self._mocker, self._project_root_path
         )
 
+    def create_contracts_cairo1(
+        self, contract_name_to_contract_dir: Dict[str, Union[str, Path]]
+    ):
+        for contract_name, contract_dir in contract_name_to_contract_dir.items():
+            assert Path(
+                contract_dir
+            ).is_dir(), "contracts in cairo1 should be represented as directories"
+            relative_path = f"src/{contract_name}"
+            shutil.copytree(
+                str(contract_dir), str(self._project_root_path / relative_path)
+            )
+        self.add_contracts_to_protostar_toml(contract_name_to_contract_dir)
```

Comment:
It's probably fine for now, but `add_contracts_to_protostar_toml` is problematic. It updates the toml but I believe it doesn't recreate Command objects created in the protostar fixture.

So the changes from `create_cairo_contracts1` will not be reflected in some methods of protostar fixture.

Reply:
2 lines below it does recreate the fixture @cptartur

Reply:
But that's a nitpick mostly

Reply:
I made `add_contracts_to_protostar_toml` private. If you have any further objections here feel free to post an issue or start a new discussion, I'm going to resolve this one for now.

Reply:
[6119836](https://github.com/software-mansion/protostar/pull/1753/commits/61198369115c971c21e693b87f161c8c74c64a21)